python_modules/dagster/dagster/core/definitions/partition.py

```diff
@@ -690,6 +692,15 @@ def get_run_config(self, partition_key: str) -> Dict[str, Any]:
             )
         return self.run_config_for_partition_fn(matching[0])
 
+    def __call__(self, *args, **kwargs):
+        if self._decorated_fn is None:
+            raise DagsterInvalidInvocationError(
+                "Only PartitionedConfig objects created using one of the partitioned config "
+                "decorators can be directly invoked."
+            )
+        else:
+            return self._decorated_fn(*args, **kwargs)
```

Comment:
we _could_ add some sort of validation step, but I'm not sure if that's even necessary / what we get out of it.